### PR TITLE
APP-492 - fp 304 404 error fix

### PR DESCRIPTION
**Pull request by nitzanpx**

### CODE REVIEW

File: lib/px/pxnginx.lua [hunk 1]
```diff
@@ -55,6 +55,10 @@ function M.application(file_name)
     local reverse_prefix = string.sub(px_config.px_appId, 3, string.len(px_config.px_appId))
     local lower_request_url = string.lower(ngx.var.request_uri)
 
+    local function is_first_party_proxy_request(reverse_prefix, lower_request_url)
+        return (px_client.reverse_px_client(reverse_prefix, lower_request_url) or px_client.reverse_px_xhr(reverse_prefix, lower_request_url) or px_client.reverse_px_captcha(reverse_prefix, lower_request_url))
```
Comment (yaronpx) [inline lib/px/pxnginx.lua]: why not just `lower_request_url`s that start with `reverse_prefix`?

File: lib/px/utils/pxclient.lua [hunk 2]
```diff
@@ -200,55 +200,101 @@ function M.load(config_file)
...
         httpc:set_keepalive()
+
+        ngx.exit(ngx.status)
+        return true
+    end
+
+    -- inteneral function, handles first party response that failed/bad status
+    -- return true for handled request
+    function default_response(content_type, content)
+        px_logger.error('Rendering default reponse on route ' .. ngx.var.uri .. 'content type: ' .. content_type .. 'body' .. content)
```
Comment (yaronpx) [inline lib/px/utils/pxclient.lua]: why are you logging this as an error? don't this have the potential of spamming the customer's log?

File: lib/px/pxnginx.lua (hunk 1 above)
Comment (nitzanpx) [reply lib/px/pxnginx.lua]: already doing it inside the functions, should we do it again?

File: lib/px/utils/pxclient.lua (hunk 2 above)
Comment (nitzanpx) [reply lib/px/utils/pxclient.lua]: 👍 
I think this might be important but we can make it a debug

File: lib/px/utils/pxclient.lua [hunk 3]
```diff
@@ -315,7 +316,7 @@ function M.load(config_file)
             ngx_req_set_header('cookie', 'pxvid=' .. vid)
         end
 
-        local status = _M.forward_to_perimeterx(px_config.collector_host, px_config.collector_port_overide)
+        local status = forward_to_perimeterx(px_config.collector_host, px_config.collector_port_overide, true )
```
Comment (yaronpx) [inline lib/px/utils/pxclient.lua]: supposed to be false

**Pull request merged**